groups/script.js

```diff
@@ -53,16 +66,24 @@ const handler = {
         });
         break;
       case 'search':
-        if (value === '') {
-          if (dataStore.groups == null) break;
-          dataStore.filteredGroupsIds = Object.values(dataStore.groups).map(
-            (group) => group.id,
+        if (featureFlags.isMultipleGroupSharingEnabled) {
+          setParamValueInURL(QUERY_PARAM_KEY.GROUP_SEARCH, value);
+          dataStore.filteredGroupsIds = getDiscordGroupIdsFromSearch(
+            Object.values(dataStore.groups),
+            value,
           );
         } else {
-          const search = value.toLowerCase();
-          dataStore.filteredGroupsIds = Object.values(dataStore.groups)
-            .filter((group) => group.title.toLowerCase().includes(search))
-            .map((group) => group.id);
+          if (value === '') {
+            if (dataStore.groups == null) break;
+            dataStore.filteredGroupsIds = Object.values(dataStore.groups).map(
+              (group) => group.id,
+            );
+          } else {
+            const search = value.toLowerCase();
+            dataStore.filteredGroupsIds = Object.values(dataStore.groups)
+              .filter((group) => group.title.toLowerCase().includes(search))
+              .map((group) => group.id);
+          }
         }
         obj[prop] = value;
         break;
```

Comment:
Can you please simplify this code as we are using too much nested component and switch case

Reply:
I made it that way because it would be easier to remove the feature flag, if the feature is working as expected we can remove the outer else block, if not remove the `if (featureFlags.isMultipleGroupSharingEnabled)` block

Reply:
I have updated the code with comment, it would be helpful while removing the feature flag